### PR TITLE
Support watch

### PR DESCRIPTION
**Pull request by WangFenjin**

### CODE REVIEW

**Comment by WangFenjin** [comment]
hi @kragniz ,

Seems the failed build doesn't related to this code change. As I'm not quite familiar with fox, please help take a look. Thanks.

File: tests/test_etcd3.py [hunk 1]
```diff
@@ -88,6 +89,90 @@ def test_delete_key(self, etcd):
         with pytest.raises(etcd3.exceptions.KeyNotFoundError):
             etcd.get('/doot/delete_this')
 
+    def test_watch_key(slef, etcd):
+        def update_etcd(v):
+            etcdctl('put', '/doot/watch', v)
+            out = etcdctl('get', '/doot/watch')
+            assert base64.b64decode(out['kvs'][0]['value']).decode('utf-8') == v
```
Comment (kragniz) [inline tests/test_etcd3.py]: All the `decode('utf-8')` calls should probably be rewritten to compare against `utils.to_bytes`

File: tests/test_etcd3.py [hunk 2]
```diff
@@ -88,6 +89,90 @@ def test_delete_key(self, etcd):
...
+
+        key_change_count = 0
+        for (event, cancel) in etcd.watch('/doot/watch'):
+            if key_change_count == 0: # first time will not have any events
+                assert event.created is True
+                assert len(event.events) == 0
+            else:
+                assert event.created is False
+                assert len(event.events) == 1
+                assert event.events[0].kv.value.decode('utf-8') == str(key_change_count)
```
Comment (kragniz) [inline tests/test_etcd3.py]: Same here, I think this should be something more similar to

    assert event.events[0].kv.value == utils.to_bytes(key_change_count)

File: tests/test_etcd3.py [hunk 3]
```diff
@@ -88,6 +89,90 @@ def test_delete_key(self, etcd):
...
+
+            # if cancel worked, we should not receive event 4
+            if len(event.events) == 1:
+                assert event.events[0].kv.value.decode('utf-8') != "4"
```
Comment (kragniz) [inline tests/test_etcd3.py]: And here, and the rest of the bytes comparisons in tests

**Comment by WangFenjin** [comment]
Hi @kragniz , all the tests passed now.

Sorry for make 2 extra commits as I don't know pep8 default ignore E126 and E226.

File: tests/test_etcd3.py [hunk 4]
```diff
@@ -88,6 +89,94 @@ def test_delete_key(self, etcd):
         with pytest.raises(etcd3.exceptions.KeyNotFoundError):
             etcd.get('/doot/delete_this')
 
+    def test_watch_key(slef, etcd):
```
Comment (yannlambret) [inline tests/test_etcd3.py]: There is a typo on 'self' for both tests

Comment (WangFenjin) [inline tests/test_etcd3.py]: Oh... Thanks @yannlambret 

It's wired they can run ...

**Comment by kragniz** [comment]
So this currently has a few issues (Ctrl-c doesn't work for cancelling a watch), but I'm just going to merge as is. Fixes can come in other pull requests. Thanks a lot for working on this @WangFenjin!

**Comment by kragniz** [comment]
See #47

**Pull request merged**